src/utilities/hoist.ts

```diff
@@ -0,0 +1,47 @@
+const REACT_LIFECYCLE_EXCLUSIONS = [
+  'setState',
+  'forceUpdate',
+  'render',
+  'componentWillMount',
+  'componentDidMount',
+  'componentWillReceiveProps',
+  'shouldComponentUpdate',
+  'componentWillUpdate',
+  'componentDidUpdate',
+  'componentWillUnmount'
+];
+
+/**
+ * Allows you to hoist methods, except those in an exclusion set from a source object into a destination object.
+ * @returns {string[]} An array of names of methods that were hoisted.
+ */
+export function hoistMethods(destination, source, exclusions = REACT_LIFECYCLE_EXCLUSIONS): string[] {
```

Comment:
it is optional at the moment.. you can use default, no exclusions, or your own exclusions..